sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -682,14 +701,14 @@ def execute(self):
     # The order of the elements is important because the inputs use
     # list indexes as references.
 
-    step_names = (
-        self._map_task.step_names or [None] * len(self._map_task.operations))
     for ix, spec in enumerate(self._map_task.operations):
       # This is used for logging and assigning names to counters.
-      operation_name = self._map_task.system_names[ix]
-      step_name = step_names[ix]
+      name_context = common.DataflowNameContext(
```

Comment:
<!--new_thread; commit:fc000934fceaaeb8c8ca87017a50b4f2cbf35340; resolved:0-->
Will we always need the Dataflow version here?

Reply:
No. The maptask will have the name contexts after internal change + PR, and this won't be necessary any longer.